Exception building library

Hello there,
I'm trying to build a "library" or "library-v3" through Gradle Task "build", but script fails with the following message:

```
FAILURE: Build failed with an exception.
* What went wrong:
A problem occurred configuring project ':library-v3'.
> groovy.lang.GroovyRuntimeException: Cannot set the value of read-only property 'executionData' for task ':library-v3:jacocoTestDebugUnitTestReport' of type org.gradle.testing.jacoco.tasks.JacocoReport.
```

Is there a way to overcome the problem?
I'm using Android Studio 4.0.2.
Thank you.
Max

--- a/build.gradle
+++ b/build.gradle
@@ -19,7 +19,7 @@ buildscript {
         google()
         jcenter()
         maven {
-            url "https://plugins.gradle.org/m2/"
+            url 'https://plugins.gradle.org/m2/'
         }
     }
     dependencies {

--- a/build.gradle
+++ b/build.gradle
@@ -18,10 +18,13 @@ buildscript {
     repositories {
         google()
         jcenter()
+        maven {
+            url "https://plugins.gradle.org/m2/"
+        }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.6.2'
-        classpath 'com.dicedmelon.gradle:jacoco-android:0.1.4'
+        classpath 'com.android.tools.build:gradle:4.0.2'
+        classpath 'com.hiya:jacoco-android:0.2'
     }
 }
 
@@ -121,13 +124,13 @@ subprojects { project ->
     if (project.ext.artifactId == null) return
 
     apply plugin: 'com.android.library'
-    apply plugin: 'jacoco-android'
+    apply plugin: 'com.hiya.jacoco-android'
     apply plugin: 'maven-publish'
     apply plugin: 'signing'
 
     // Code coverage
     jacoco {
-        toolVersion = "0.8.5"
+        toolVersion = "0.2"
     }
 
     tasks.withType(Test) {

--- a/build.gradle
+++ b/build.gradle
@@ -130,7 +130,7 @@ subprojects { project ->
 
     // Code coverage
     jacoco {
-        toolVersion = "0.2"
+        toolVersion = "0.8.4"
     }
 
     tasks.withType(Test) {

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,6 @@
+#Mon Nov 02 14:06:20 EST 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip

